src/app/application/components/data-collection/DataCollection.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <slot
+    v-if="items.length === 0"
+    name="empty"
+    :items="items"
+  >
+    <EmptyBlock />
+  </slot>
+  <slot
+    v-else
+    name="default"
+    :items="paginated"
+  />
+
+  <slot
+    v-if="props.page !== 0 && items.length > 0"
+    name="pagination"
+    :items="paginated"
+  >
+    <KPagination
+      :total-count="items.length"
+      :current-page="props.page"
+      :initial-page-size="props.pageSize"
+      :page-sizes="[15, 30, 50, 75, 100]"
+      @page-changed="({ page }: PaginationChangeEvent) => {
+        emit('change', {
+          page,
+          pageSize: props.pageSize,
+        })
+      }"
+    />
+  </slot>
```

Comment:
Nowhere

>  (and pagination outside of a AppCollection table should we need that)

I want this component to be built in a way that makes sure it thinks about pagination for if we need to paginate this view (outbound lists). Therefore I built it with that in mind. Now if we need to paginate the outbound listing if it has 100's of inbounds, its pretty much done.

Reply:
Shouldn’t we only add this when we actually need it? As far as I know, the designs don’t include pagination for this either. This seems like premature optimization.

Reply:
Premature optimization would be building something to be more optimized when it thats not necessary.

This isn't about optimization its about making sure that DataCollection is built in a way that we can use it for paging if we ever need to. I don't want to have to rebuild it all completely differently if someone says "can we add pagination to this non-table thing please?"

Reply:
Or even: I don't want to have to rebuild it all completely differently if someone says "can we add frontend pagination to this table thing please?"

Reply:
Aside from the pedantry about whether I’m using the right term here, we don’t currently have concrete plans to use this bit of code. It’s doubtful having to "rebuild it all completely differently" would become necessary in the event that we end up adding pagination here.

Looking past pagination, this component offers conditionally rendering an EmptyBlock and no functionality beyond that.

So I’m kind of confused: Why do we need this right now? It doesn’t seem the right time to introduce an abstraction over the search/filtering capabilities that are yet to be added in this view.

Reply:
> Aside from the pedantry about whether I’m using the right term here

There's a big difference between building something so its flexible for the long term (not knowing what features might be required in the future) and micro and/or premature optimization, as long as you know making sure this worked alongside KPagination isn't about optimizing performance.

Happy to remove/comment out the KPagination bit now the component has been around it. lmk

> Why do we need this right now?

Well one more thing is it means I need to do less of `<template v-for>` which I know you don't particularly like.

Reply:
> It doesn’t seem the right time to introduce an abstraction over the search/filtering capabilities that are yet to be added in this view.

We filter by `inactive` in this view

Reply:
This is definitely much less about the introduction of a new component than about the addition of features that we’re not using. Since this PR contains work that was deemed relevant for the upcoming release of Kuma, it would’ve made more sense to keep it slim and isolated to those feature additions.

Nonetheless, I’m going to resolve this thread now so we can get this in.